bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1128 @@
+                            }
+                        });
+                    }
+                }
+            });
+            legacyLinkedChannelUIDs.clear();
+        }
+    }
+
+    /**
+     * Set the active list of channels by removing any that had initially been created by the thing XML declaration, but
+     * which in fact did not have data returned from the bridge i.e. channels which are not in the supportedChannelIdSet
+     *
+     * Also warn if there are channels in the supportedChannelIdSet set which are not in the thing.
+     *
+     * Adjusts the channel list so that only the highest level channel is available in the normal channel list. If a
+     * light supports the color channel, then it's brightness and switch can be commanded via the 'B' part of the HSB
+     * channel value. And if it supports the brightness channel the switch can be controlled via the brightness. So we
+     * can remove these lower level channels from the normal channel list.
+     *
+     * For more advanced applications, it is necessary to orthogonally command the color xy parameter, dimming
+     * parameter, and/or on/off parameter independently. So we add corresponding advanced level 'CHANNEL_2_BLAH_ONLY'
+     * channels for that purpose. Since they are advanced level, normal users should normally not be confused by them,
+     * yet advanced users can use them nevertheless.
+     */
+    private void updateChannelList() {
+        if (!disposing) {
+            synchronized (supportedChannelIdSet) {
+                logger.debug("{} -> updateChannelList()", resourceId);
+
+                if (supportedChannelIdSet.contains(CHANNEL_2_COLOR)) {
+                    supportedChannelIdSet.add(CHANNEL_2_COLOR_XY_ONLY);
+                    //
+                    supportedChannelIdSet.remove(CHANNEL_2_BRIGHTNESS);
+                    supportedChannelIdSet.add(CHANNEL_2_DIMMING_ONLY);
+                    //
+                    supportedChannelIdSet.remove(CHANNEL_2_SWITCH);
+                    supportedChannelIdSet.add(CHANNEL_2_ON_OFF_ONLY);
+                }
+                if (supportedChannelIdSet.contains(CHANNEL_2_BRIGHTNESS)) {
+                    supportedChannelIdSet.add(CHANNEL_2_DIMMING_ONLY);
+                    //
+                    supportedChannelIdSet.remove(CHANNEL_2_SWITCH);
+                    supportedChannelIdSet.add(CHANNEL_2_ON_OFF_ONLY);
+                }
+                if (supportedChannelIdSet.contains(CHANNEL_2_SWITCH)) {
+                    supportedChannelIdSet.add(CHANNEL_2_ON_OFF_ONLY);
+                }
+
+                // warn about any missing channels
+                supportedChannelIdSet.stream().filter(channelId -> Objects.isNull(thing.getChannel(channelId)))
+                        .forEach(channelId -> logger.warn(
+                                "{} -> updateChannelList() required channel '{}' missing => please recreate thing!",
+                                resourceId, channelId));
```

Comment:
Just for my understanding, is this warning only relevant for testers of this binding version (including community version) who created things prior to these channels being added?

Reply:
> only relevant for testers

Mostly for such people. But I have also had experience with dynamically created channels (not only in this binding) where sometimes the framework might 'lose' a channel. For whatever reason. It is highly unlikely. But therefore probably worth a warning in the log just in case..

Reply:
Consider adding a comment for this. I think this should be removed in the future.

Reply:
Ok.